src/EnergyPlus/ScheduleManager.cc

```diff
@@ -3030,15 +3060,35 @@ namespace ScheduleManager {
 				EMin = MMField;
 			}
 
+			if ( interpolationKind == ScheduleInterpolation::Linear ) {
+				totalMinutes = ( EHr - SHr ) * 60 + ( EMin - SMin ) + 1;
+				if ( totalMinutes == 0 ) totalMinutes = 1; // protect future division
+				if ( Count == 1 ) {
+					StartValue = Numbers( Count ); //assume first period is flat
+					EndValue = Numbers( Count );
+				} else {
+					StartValue = EndValue;
+					EndValue = Numbers( Count );
+				}
+				incrementPerMinute = ( EndValue - StartValue ) / totalMinutes;
+				curValue = StartValue + incrementPerMinute;
+			}
+
 			if ( SHr == EHr ) {
 				for ( Min = SMin; Min <= EMin; ++Min ) {
 					if ( SetMinuteValue( Min, SHr ) ) {
 						ShowSevereError( "ProcessScheduleInput: ProcessIntervalFields, Processing time fields, overlapping times detected, " + ErrContext + '=' + DayScheduleName );
 						ErrorsFound = true;
 						goto UntilLoop_exit;
 					}
-					MinuteValue( Min, SHr ) = Numbers( Count );
-					SetMinuteValue( Min, SHr ) = true;
+					if ( interpolationKind == ScheduleInterpolation::Linear ) {
+						MinuteValue( Min, EHr ) = curValue;
+						curValue += incrementPerMinute;
```

Comment:
I see that this block **is** inside a minute loop, so I get why it is only incrementing by incrementPerMinute each step.  Still don't see why it is doing it above.  Maybe it's just because of the diff view and I'm missing something not shown.

Reply:
It is the value at the first minute. If curValue started as StartValue than minute zero (at the top of the hour) and minute one (one minute after the hour) would be the same value.